src/main/java/com/google/sps/servlets/YoutubeApi.java

```diff
@@ -51,8 +60,36 @@ private static YouTube getService() throws GeneralSecurityException, IOException
       VideoListResponse response =
           getService().videos().list("snippet").setKey(DEVELOPER_KEY).setId(id).execute();
       Video video = response.getItems().get(0);
-      YoutubePost newPost = new YoutubePost(
-          video.getSnippet().getTitle(), video.getSnippet().getDescription(), video.getId());
+      /**Retrieving comments before creating new post */
+
+      YouTube.CommentThreads.List request = getService()
+                                                .commentThreads()
+                                                .list("snippet")
+                                                .setVideoId(video.getId())
+                                                .setTextFormat("plainText");
+      CommentThreadListResponse commentsResponse =
+          request.setKey(DEVELOPER_KEY).setMaxResults(15L).execute();
+      List<String> plainComments = new ArrayList<String>();
+      List<Long> commentLikesList = new ArrayList<Long>();
+      Long commentLikes = 0L;
+      if (commentsResponse == null) {
```

Comment:
Can the response be null? I would think if there are no comments, the response would be empty but not null

Reply:
Youre right its a list so the list would just be empty